qiskit/pulse/device_specification.py

```diff
@@ -0,0 +1,93 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2019, IBM.
+#
+# This source code is licensed under the Apache License, Version 2.0 found in
+# the LICENSE.txt file in the root directory of this source tree.
+
+"""
+Specification of the device.
+"""
+import logging
+from typing import List
+
+from qiskit.pulse.qubit import Qubit
+from qiskit.pulse.channels import DriveChannel, ControlChannel, MeasureChannel
+from qiskit.pulse.channels import AcquireChannel, MemorySlot, RegisterSlot
+
+logger = logging.getLogger(__name__)
+
+
+class DeviceSpecification:
```

Comment:
This should not be a fundamental object within a `Schedule`, but a helper object to aid in the programming of the device by accessing channels to apply and append schedules too.